### PR TITLE
Cleaning up error messages and removing unused statetracker states

### PR DESCRIPTION
**Pull request by khandpur**
- Clean up error messages
- Remove unused statetracker states
- fix for a bug that applied search action one last time just before a timeout without checking the hierarchy again. as a result the screenshot was different from when the actual failure was expected to occur.

### CODE REVIEW

File: Tests/FunctionalTests/Sources/FTRScrollViewTest.m [hunk 1]
```diff
@@ -340,7 +340,7 @@ - (GREYElementMatcherBlock *)matcherTextWithCGPointValue:(CGPoint)expected
     return ABS(actual.x - expected.x) <= accuracy && ABS(actual.y - expected.y) <= accuracy;
   };
   return [GREYElementMatcherBlock matcherWithMatchesBlock:matchesValueWithAccuracy
-                                         descriptionBlock:^(id<GREYDescription> description) {
+                                       descriptionBlock:^(id<GREYDescription> description) {
```
Comment (sid-github) [inline Tests/FunctionalTests/Sources/FTRScrollViewTest.m]: Why was this changed?

Comment (khandpur) [inline Tests/FunctionalTests/Sources/FTRScrollViewTest.m]: Oops - I'll revert it.

**Comment by khandpur** [comment]
PTAL
